src/js/common/components/CollapsiblePanel.jsx

```diff
@@ -0,0 +1,72 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Scroll from 'react-scroll';
+import _ from 'lodash/fp';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+export default class CollapsiblePanel extends React.Component {
+  constructor() {
+    super();
+    this.toggleChapter = this.toggleChapter.bind(this);
+    this.state = { open: false };
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId();
+  }
+
+  scrollToTop() {
+    scroller.scrollTo(`collapsiblePanel-${this.id}-ScrollElement`, {
```

Comment:
nit: just hyphenate all of these instead of mixing hyphenated-names with camelCase?